elasticdl/python/data/data_reader.py

```diff
@@ -78,18 +90,28 @@ def create_shards(self):
     def records_output_types(self):
         return tf.string
 
+    @property
+    def metadata(self):
+        return Metadata(column_names=None)
+
 
 class ODPSDataReader(AbstractDataReader):
     def __init__(self, **kwargs):
         AbstractDataReader.__init__(self, **kwargs)
         self._kwargs = kwargs
+        self._metadata = Metadata(column_names=None)
 
     def read_records(self, task):
         reader = self._get_reader(
             table_name=self._get_odps_table_name(task.shard_name)
         )
+        if self._metadata.column_names is None:
```

Comment:
So, if `_metadata.colume_names` is not None, but different from `get("columns")`, do we do any validation here?

Reply:
No we don't need validation here in this case since `self._kwargs.get("columns")` won't get changed at all (this is only specified in the constructor).